### DISCUSSION TITLE
[BUG]

### DISCUSSION THREAD

**Discussion by yashhjaggi1998** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1181] [Q&A]
**Describe the bug**

I am trying to start camera stream for 3 cameras using a single raspberry pi 4 device. I am using the following command in my .sh script: 

**kvs_gstreamer_sample ${CAMERA_NAMES[$index]} $RTSP_URL ! videoconvert ! videoscale ! video/x-raw,width=320,height=240 ! queue ! x264enc bitrate=500 speed-preset=superfast tune=zerolatency ! video/x-h264,profile=baseline ! h264parse ! video/x-h264,stream-format=avc**

The problem I am facing is sometimes all 3 cameras publish stream to KVS, sometimes its 2 and sometimes just 1. 

There is no consistency. The cameras work fine because when I try to publish them individually they stream video just fine. For 3 camera I am running the above command in my shell shell script using the background process '&' in shell script. I beleiev I am running on 3 cameras on single core. 

I am seeking your help in order to publish video from all 3 cameras consistently.

**Platform (please complete the following information):**
 - Raspberry pi 4

**Comment by disa6302** [comment]
@yashhjaggi1998 ,

Could you provide VERBOSE logs? What does the memory consumption look like when you try to run 3 cameras on a single core?

**Reply by yashhjaggi1998** [reply]
How do I get the verbose logs? 

As for the memory consumption, would a screenshot of "htop" command work? I can also share the camera specifications with you since I believe the lowest video resolution supported by my camera is 1240x1080.
